src/services/regenerate-document-task-processor.spec.js

```diff
@@ -0,0 +1,56 @@
+import sinon from 'sinon';
+import uniqueId from '../utils/unique-id.js';
+import DocumentService from './document-service.js';
+import { setupTestEnvironment, destroyTestEnvironment } from '../test-helper.js';
+import RegenerateDocumentTaskProcessor from './regenerate-document-task-processor.js';
+
+describe('RegenerateDocumentTaskProcessor', () => {
+  let container;
+  let documentService;
+  const sandbox = sinon.createSandbox();
+
+  let sut;
+
+  beforeAll(async () => {
+    container = await setupTestEnvironment();
+    documentService = container.get(DocumentService);
+    sut = container.get(RegenerateDocumentTaskProcessor);
+  });
+
+  afterAll(async () => {
+    await destroyTestEnvironment(container);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('process', () => {
+    beforeEach(() => {
+      sandbox.stub(documentService, 'regenerateDocument');
+    });
```

Comment:
This could go in the outer most `describe`, with its matching `afterEach`, I think it's better if all stubbing is done at root level (unless it's an occasional stub, where you'd want the real functionality in some cases, which is rare in unit tests).

I find that gathering all stubbing to the top simplifies the setup of inner tests and reduces confusion or double stubbing.

Reply:
Got it, fixed, thanks